src/tyctx.rs

```diff
@@ -78,9 +80,13 @@ impl TyCtx {
                 span: ident.span.variant(span_variant),
             };
             if !self.declarations.borrow().contains_key(&new_ident) {
-                break new_ident;
+                return new_ident;
             }
-        };
+        }
+    }
+
+    pub fn fresh_var(&self, ident: Ident, span_variant: SpanVariant) -> Ident {
```

Comment:
Please add documentation to distinguish `fresh_ident` and `fresh_var`, i.e. that the latter assumes an existing variable declaration and creates a new one automatically with a name from `fresh_ident`.